packages/read-model/src/Image/ImageViewRepository.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Image;
+
+use Shopsys\FrameworkBundle\Component\Image\ImageFacade;
+
+/**
+ * @experimental
+ */
+class ImageViewRepository
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\ImageFacade
+     */
+    protected $imageFacade;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Image\ImageViewFactory
+     */
+    protected $imageViewFactory;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Image\ImageFacade $imageFacade
+     * @param \Shopsys\ReadModelBundle\Image\ImageViewFactory $imageViewFactory
+     */
+    public function __construct(ImageFacade $imageFacade, ImageViewFactory $imageViewFactory)
+    {
+        $this->imageFacade = $imageFacade;
+        $this->imageViewFactory = $imageViewFactory;
+    }
+
+    /**
+     * @param string $entityClass
+     * @param array $entityIds
+     * @return array
+     */
+    public function getForEntityIds(string $entityClass, array $entityIds): array
+    {
+        $imagesIndexedByEntityIds = $this->imageFacade->getImagesOrNullsByEntitiesIndexedByEntityId($entityIds, $entityClass);
+
+        $imageViewsOrNullsIndexedByEntityIds = [];
+        foreach ($entityIds as $entityId) {
+            if (!isset($imagesIndexedByEntityIds[$entityId])) {
```

Comment:
```suggestion
            $imageViewsOrNullsIndexedByEntityIds[$entityId] = $this->createView($imagesIndexedByEntityIds[$entityId]);
            ...
            protected function createView(?Image $image): ?ImageView {
                if ($image === null) {
                    return null;
                }
                return $this->imageViewFactory->createFromImage($image);
            }
```

the whole `if` is difficult to understand for me, so I suggest to change it to a single call of function and deal with details inside

Reply:
I had to rewrite it a little bit more because I realized `getImagesOrNullsByEntitiesIndexedByEntityId` does not do what claims to do. It's not returning null values for not existing ids. 
Please check it out.

Reply:
@grossmannmartin oh, my apologies, I have rewritten the function but not its name :man_facepalming:

Reply:
seems great :+1: